voip_oca/models/mail_activity.py

```diff
@@ -40,5 +40,12 @@ def get_call_activities(self, _search, offset, limit):
                 [[(field, "ilike", _search)] for field in search_fields]
             )
             domain = expression.AND([domain, search_domain])
-        activities = self.search(domain, offset=offset, limit=limit)
-        return activities.activity_format()
+        all_activities = self.search(domain, offset=offset, limit=limit)
+        # Filter activities to avoid accessing records that the user cannot read
+        # due to multi-company restrictions or other access rules in the res_model.
+        allowed_activities = []
```

Comment:
Please name variable properly:

```suggestion
        allowed_activity_ids = []
```